server/apps/packages/package_service.py

```diff
@@ -272,6 +272,19 @@ def replace_ref_in_package(self, package, old_ref_id, new_ref_id):
                 g[ITEM_REF] = new_ref_id
                 g['guid'] = new_ref_id
 
+    def update_field_in_package(self, package, ref_id, field, field_value):
+        """
+        Locates the reference with the ref_id and replaces field value
+        :param package: Package
+        :param ref_id: reference id
+        :param field: field to be replaced
+        :param field_value: value to be used
+        """
+        non_root_groups = (group for group in package.get(GROUPS, []) if group.get(GROUP_ID) != ROOT_GROUP)
```

Comment:
That's also an option indeed, although we need to be careful that readability does not suffer. The latter might happen, for example, when trying to pack a long long and/or complex comprehension, but it needs to be evaluated on a case to case basis. Using an intermediary variable with a meaningful name is a good strategy to fight that

I'm kind of more inclined to use the latter approach in this case, makes the intent more clear IMO.